app/collections/collaborators.jsx

```diff
@@ -0,0 +1,375 @@
+
+      return (
+        <div className="confirm-delete-dialog content-container">
+          <p>Are you sure you want to delete this role?</p>
+          <div>
+            <button className="minor-button" autoFocus={true} onClick={resolve}>No</button>
+            {' '}
+            <button className="major-button" onClick={handleDelete}>Yes</button>
+          </div>
+        </div>
+      );
+    });
+  }
+
+  updateRoles(newRoles = [], callback = () => {}) {
+    let promise;
+    this.setState({ saving: true });
+
+    if (newRoles.length > 0) {
+      promise = this.props.roleSet.update({ roles: newRoles }).save();
+    } else {
+      promise = this.props.roleSet.delete();
+    }
+
+    promise.then(() => {
+      callback();
+      this.setState({ saving: false });
+    })
+    .catch((error) => {
+      this.setState({ error });
+    });
+  }
+
+  render() {
+    const { owner } = this.state;
+
+    return (
+      <p>
+        {owner && (<strong>{owner.login}</strong>)}
+
+        { owner.display_name !== this.props.collectionOwner && (
+          <button type="button" className="pill-button" onClick={this.removeRoles}>Remove</button>
+        )}
+
+        <span className="columns-container inline">
+          {POSSIBLE_ROLES.map((role) => {
+            const checkedRole = this.props.roleSet.roles.includes(role);
+            const boldRole = checkedRole ? <strong>{ROLES_INFO[role].label}</strong> : ROLES_INFO[role].label;
+            const disableCheck = role === 'owner' || this.state.saving || owner.display_name === this.props.collectionOwner;
+
+            return (
+              <label htmlFor="role" key={role}>
+                <input type="checkbox" name={role} checked={checkedRole} disabled={disableCheck} onChange={this.toggleRole.bind(this, role)} />{' '}
+                {boldRole}
+              </label>
+            );
+          })}
+        </span>
+      </p>
+    );
+  }
+}
+
+RoleRow.propTypes = {
+  collectionOwner: React.PropTypes.string,
+  onRemove: React.PropTypes.func,
+  roleSet: React.PropTypes.shape({
+    delete: React.PropTypes.func,
+    get: React.PropTypes.func,
+    roles: React.PropTypes.array,
+    update: React.PropTypes.func
+  })
+};
+
+export class CollectionCollaborators extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      error: null,
+      hasSettingsRole: false,
+      owner: null,
+      roleSets: []
+    };
+
+    this.update = this.update.bind(this);
+    this.handleCollaboratorChange = this.handleCollaboratorChange.bind(this);
+  }
+
+  componentDidMount() {
+    checkIfCollectionOwner(this.props.user, this.props.collection)
+    .then((hasSettingsRole) => {
+      this.setState({ hasSettingsRole });
+    });
+    this.update();
+  }
+
+  update(callback = () => {}) {
+    const promise = Promise.all([
+      apiClient.type('collection_roles').get({ collection_id: this.props.collection.id }),
+      this.props.collection.get('owner')
```

Comment:
The collection owner is already fetched for in the parent `CollectionPage`. Pass it down as a prop instead of fetching it again.